lib/charms/opensearch/v0/opensearch_secrets.py

```diff
@@ -58,8 +58,16 @@ def _on_secret_changed(self, event: ActionEvent):
         """Refresh secret and re-run corresponding actions if needed."""
         if not event.secret.label:
             logger.info("Secret %s has no label, ignoring it.", event.secret.id)
+            return
+
+        try:
+            label_parts = self.breakdown_label(event.secret.label)
+        except ValueError:
+            logging.info(f"Label {event.secret.label} was meaningless for us, returning")
+            return
+        if not label_parts:
+            return
```

Comment:
This can be removed. `label_parts` cannot be None.

Reply:
Yes it can. If it's an external secert ;-)

Reply:
That's the whole point of the change... We do not only have internal secerts anymore...

Reply:
`label_parts = self.breakdown_label(..)` - [this can either](https://github.com/canonical/opensearch-operator/blob/873136f527413cc874ee1a0c414f7eff1fd75bfe/lib/charms/opensearch/v0/opensearch_secrets.py#L97) raise a `ValueError` - or return a Dict, not a `None` value  🙂

Reply:
We were writing simultaneously.

Yes you're right I remove it and see if all good on AWS